blackbird_python/blackbird/listener.py

```diff
@@ -320,16 +323,21 @@ def exitArrayvar(self, ctx: blackbirdParser.ArrayvarContext):
             shape = tuple([int(i) for i in ctx.shape().getText().split(",")])
 
         value = []
+        parameters = []
+        rows = 0
         # loop through all children of the 'arrayval' branch
         for i in ctx.arrayval().getChildren():
             # Check if the child is an array row (this is to
             # avoid the '\n' row delimiter)
             if isinstance(i, blackbirdParser.ArrayrowContext):
-                value.append([])
```

Comment:
Not quite sure what the purpose of this one was before :thinking: How come it could be removed?

Reply:
I changed the way that arrays are read; before, they created a nested list (dim=2) from start, but because I needed to insert parameters in it, it was easier to keep it one-dimensional here and then reshape it later.

Reply:
Might be worth adding your comment above as a comment in the code @thisac, it definitely helps to understand the code logic